integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/OpenHouseTableOperations.java

```diff
@@ -276,9 +276,16 @@ static Mono<GetTableResponseBody> handleCreateUpdateHttpError(
       return Mono.error(
           new CommitFailedException(
               casted, casted.getStatusCode().value() + " , " + casted.getResponseBodyAsString()));
+
     } else if (e instanceof WebClientResponseException.GatewayTimeout
-        | e instanceof WebClientResponseException.ServiceUnavailable) {
-      return Mono.error(new CommitStateUnknownException(e));
+        || e instanceof WebClientResponseException.ServiceUnavailable
+        || e instanceof WebClientResponseException.InternalServerError) {
```

Comment:
This PR is to catch InternalServerError as CommitStateUnknownException, but it should have already been captured in the else block?

Reply:
InternalServerError comes as WebClientResponseException still and hence would have been returned as WebCientResponseWithMessageException  in 
else if (e instanceof WebClientResponseException) block. So it would not have been treated as CommitStateUnknownException

Reply:
Gotcha.